extensions/data-transfer/portability-data-transfer-backblaze/src/test/java/org/datatransferproject/datatransfer/backblaze/common/BackblazeDataTransferClientTest.java

```diff
@@ -53,192 +57,180 @@
+  }
+
+  private void createEmptyBucketList() {
+    when(s3Client.listBuckets()).thenReturn(ListBucketsResponse.builder().build());
+  }
+
+  private BackblazeDataTransferClient createDefaultClient() {
+    return new BackblazeDataTransferClient(monitor, backblazeS3ClientFactory, 1000, 500);
+  }
+
+  @Test
+  public void testWrongPartSize() {
+    assertThrows(IllegalArgumentException.class, () -> {
+      new BackblazeDataTransferClient(monitor, backblazeS3ClientFactory, 10, 0);
+    });
+  }
+
+  @Test
+  public void testInitBucketNameMatches() throws BackblazeCredentialsException, IOException {
+    createValidBucketList();
+    BackblazeDataTransferClient client = createDefaultClient();
+    client.init(KEY_ID, APP_KEY, EXPORT_SERVICE);
+    verify(s3Client, times(0)).createBucket(any(CreateBucketRequest.class));
+  }
+
+  @Test
+  public void testInitBucketCreated() throws BackblazeCredentialsException, IOException {
+    Bucket bucket = Bucket.builder().name("invalid-name").build();
+    when(s3Client.listBuckets()).thenReturn(ListBucketsResponse.builder().buckets(bucket).build());
+    BackblazeDataTransferClient client = createDefaultClient();
+    client.init(KEY_ID, APP_KEY, EXPORT_SERVICE);
+    verify(s3Client, times(1)).createBucket(any(CreateBucketRequest.class));
+  }
+
+  @Test
+  public void testInitBucketNameExists() throws BackblazeCredentialsException, IOException {
+    createEmptyBucketList();
+    when(s3Client.createBucket(any(CreateBucketRequest.class)))
+        .thenThrow(BucketAlreadyExistsException.builder().build());
+    BackblazeDataTransferClient client = createDefaultClient();
+
+    assertThrows(IOException.class, () -> {
+      try {
         client.init(KEY_ID, APP_KEY, EXPORT_SERVICE);
-        // Act
-        String actualVersionId = client.uploadFile(FILE_KEY, testFile);
-        // Assert: uploaded in 8 parts
-        verify(s3Client, times((int) expectedParts))
-                .uploadPart(any(UploadPartRequest.class), any(RequestBody.class));
-        assertEquals(expectedVersionId, actualVersionId);
-    }
-
-    @Test(expected = IOException.class)
-    public void testUploadFileMultipartException() throws BackblazeCredentialsException, IOException {
-        // Arrange
-        createValidBucketList();
-        when(s3Client.createMultipartUpload(any(CreateMultipartUploadRequest.class)))
-                .thenReturn(CreateMultipartUploadResponse.builder().uploadId("xyz").build());
-        when(s3Client.uploadPart(any(UploadPartRequest.class), any(RequestBody.class)))
-                .thenThrow(AwsServiceException.builder().build());
-        final long fileSize = testFile.length();
-        BackblazeDataTransferClient client =
-                new BackblazeDataTransferClient(monitor, backblazeS3ClientFactory, fileSize / 2, fileSize / 8);
+      } catch (Exception ex) {
+        verify(monitor, atLeast(1)).info(any());
+        throw ex;
+      }
+    });
```

Comment:
Could you please remove the try-catch block and move monitor verification outside? Sorry I missed it during the first review :(